src/include/common/cpu/rpp_cpu_rgb_hsv_conversion.hpp

```diff
@@ -0,0 +1,220 @@
+    pMask[2] = _mm256_and_ps(pMask[0], pMask[1]);                                                                      //     if (cmax == rf)
+    pH = _mm256_and_ps(pMask[2], _mm256_sub_ps(pVecG, pVecB));                                                         //         hue = gf - bf;
+    pAdd = _mm256_and_ps(pMask[2], avx_p0);                                                                            //         add = 0.0f;
+    pMask[3] = _mm256_cmp_ps(pV, pVecG, _CMP_EQ_OQ);                                                                   //     Temporarily store cmax == gf comparison
+    pMask[2] = _mm256_andnot_ps(pMask[1], pMask[3]);                                                                   //     else if (cmax == gf)
+    pH = _mm256_or_ps(_mm256_andnot_ps(pMask[2], pH), _mm256_and_ps(pMask[2], _mm256_sub_ps(pVecB, pVecR)));           //         hue = bf - rf;
+    pAdd = _mm256_or_ps(_mm256_andnot_ps(pMask[2], pAdd), _mm256_and_ps(pMask[2], avx_p2));                            //         add = 2.0f;
+    pMask[3] = _mm256_andnot_ps(pMask[3], _mm256_andnot_ps(pMask[1], pMask[0]));                                       //     else
+    pH = _mm256_or_ps(_mm256_andnot_ps(pMask[3], pH), _mm256_and_ps(pMask[3], _mm256_sub_ps(pVecR, pVecG)));           //         hue = rf - gf;
+    pAdd = _mm256_or_ps(_mm256_andnot_ps(pMask[3], pAdd), _mm256_and_ps(pMask[3], avx_p4));                            //         add = 4.0f;
+    pH = _mm256_or_ps(_mm256_andnot_ps(pMask[0], pH), _mm256_and_ps(pMask[0], _mm256_div_ps(pH, pDelta)));             //     hue /= delta; }}
+    pH = _mm256_add_ps(pH, pAdd);                                                                                      //     hue += add;
+}
+
+// Converts RGB to HSV color space for a single pixel (scalar version)
+inline void RGB_to_HSV(RpptFloatRGB *pixel, Rpp32f &hue, Rpp32f &sat, Rpp32f &val)
+{
+    // Find maximum and minimum values among RGB components
+    Rpp32f rf = pixel->R, gf = pixel->G, bf = pixel->B;
+    Rpp32f cmax = RPPMAX3(rf, gf, bf);
+    Rpp32f cmin = RPPMIN3(rf, gf, bf);
+    Rpp32f delta = cmax - cmin;
+
+    // Initialize HSV values
+    hue = 0.0f;
+    sat = 0.0f;
+    val = cmax;
+
+    // Calculate saturation and hue if delta is not zero and max value is not zero
+    if ((delta != 0) && (cmax != 0)) {
+        sat = delta / cmax;
+        // Calculate hue based on which RGB component is maximum
+        if (cmax == rf)
+        {
+            hue = (gf - bf) / delta;
+        }
+        else if (cmax == gf) {
+            sat = delta / cmax;
+            hue = (bf - rf) / delta + 2.0f;
+        } else {
+            sat = delta / cmax;
+            hue = (rf - gf) / delta + 4.0f;
+        }
+    }
+}
+
+// Converts HSV to RGB color space for a single pixel (scalar version)
+inline void HSV_to_RGB(Rpp32f hue, Rpp32f sat, Rpp32f val, RpptFloatRGB *pixel)
```

Comment:
The raw C and AVX versions are inconsistent in not passing pointers or using call by reference. Pls use call by reference